File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const n = prices.length;
    const answer = Array(n).fill(0);
    
    for(let i = 0; i<n; i++){
        let time = 0;
        for(let j = i+1; j < n; j++){
            time++;
            if(prices[i] > prices[j]){
                break;
            }
        }
        answer[i] = time;
    }
    
    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0] * n
    stack = []
    
    for i, price in enumerate(prices):
        while len(stack) > 0 and prices[stack[-1]] > price:
            j = stack.pop()
            answer[j] = i - j
        
        stack.append(i)
        
        
    while len(stack) > 0:
        idx = stack.pop()
        answer[idx] = n - 1 - idx
        
    return answer